torch/csrc/jit/passes/onnx.cpp

```diff
@@ -26,6 +26,12 @@ bool hasUsedHandle(Node *node) {
 
 // Transform PythonOps and Cpp Ops into Node's that match ONNX semantics.
 void ToONNX(std::shared_ptr<tracer::TracingState>& state) {
+  // Check that the tracing state is live (it should be, because
+  // you were supposed to request zero derivatives.)
+  if (state->is_expired()) {
+    throw std::runtime_error("Tracing state is expired!  You should run the tracer with num_derivatives=0");
```

Comment:
No, `num_derivatives=0` is not specific to `ToONNX` pass in any way. We use it e.g. when running `word_language_model` example with `num_derivatives=1`

Reply:
OK, I'm going to do something like:

```
diff --git a/torch/csrc/jit/passes/onnx.cpp b/torch/csrc/jit/passes/onnx.cpp
index 842d79f..0596d05 100644
--- a/torch/csrc/jit/passes/onnx.cpp
+++ b/torch/csrc/jit/passes/onnx.cpp
@@ -29,7 +29,7 @@ void ToONNX(std::shared_ptr<tracer::TracingState>& state) {
   // Check that the tracing state is live (it should be, because
   // you were supposed to request zero derivatives.)
   if (state->is_expired()) {
-    throw std::runtime_error("Tracing state is expired!  You should run the tracer with num_derivatives=0");
+    throw std::logic_error("ToONNX: tracing state is expired");
   }
 
   auto new_graph = std::make_shared<Graph>();
```